### DISCUSSION TITLE
What to do in Emacs 29?

### DISCUSSION THREAD

**Discussion by refaelsh** [emacs-tree-sitter/elisp-tree-sitter#270] [Q&A] [answered]
The README.md simply states:
```
For Emacs 29+, please use [the built-in integration](https://lists.gnu.org/archive/html/emacs-devel/2022-11/msg01443.html) instead of this package.
```
But when I follow the link, I only see the announcement of Tree-sitter being integrated into Emacs 29.
I expected a link to documentation explaining what to do now that we all have Emacs 29+.

**Comment by sogaiu** [comment]
May be other folks can suggest better resources, but for the moment, perhaps the following might be better than nothing?

* https://www.masteringemacs.org/article/how-to-get-started-tree-sitter
* https://blog.markhepburn.com/posts/experimenting-with-the-built-in-treesitter-support-in-emacs/

**Reply by refaelsh** [reply]
Yes, I've seen both of those links. But:
1. Both of them are not the official documentation.
2. Both of them from 2022.

**Reply by refaelsh** [reply]
For example, Eglot, is also a new feature that's been integrated into Emacs in 29. But it has official documentation: https://www.gnu.org/software/emacs/manual/html_mono/eglot.html.

**Reply by sogaiu** [reply, marked as answer]
If you're looking for developer-oriented docs (e.g. for making a major mode with tree-sitter support), there is [this](https://github.com/emacs-mirror/emacs/blob/ef01b634d219bcceda17dcd61024c7a12173b88c/admin/notes/tree-sitter/starter-guide), though I'm not sure how up-to-date it is.

There is also [this](https://www.gnu.org/software/emacs/manual/html_node/elisp/Parsing-Program-Source.html) in the Emacs Lisp docs and some stuff in the official manual like [this](https://www.gnu.org/software/emacs/manual/html_node/emacs/Parser_002dbased-Font-Lock.html).  Searching through stuff under [here](https://github.com/emacs-mirror/emacs/tree/ef01b634d219bcceda17dcd61024c7a12173b88c/doc) might turn up more.

If there aren't many responses here before long, may be you'd have more luck asking at the [emacs devel mailing list](https://lists.gnu.org/archive/html/emacs-devel/) or searching the associated archives?

**Reply by refaelsh** [reply]
> may be you'd have more luck asking at the [emacs devel mailing list](https://lists.gnu.org/archive/html/emacs-devel/)

I agree.
